vividus-plugin-aws-lambda/src/main/java/org/vividus/aws/lambda/steps/LambdaSteps.java

```diff
@@ -0,0 +1,104 @@
+     *                     encoding on
+     *                     this field before sending this request to the AWS service. Users should not perform Base64
+     *                     encoding on this field.
+     * @param scopes       The set (comma separated list of scopes e.g.: STORY, NEXT_BATCHES) of variables scopes<br>
+     *                     <i>Available scopes:</i>
+     *                     <ul>
+     *                     <li><b>STEP</b> - the variable will be available only within the step,
+     *                     <li><b>SCENARIO</b> - the variable will be available only within the scenario,
+     *                     <li><b>STORY</b> - the variable will be available within the whole story,
+     *                     <li><b>NEXT_BATCHES</b> - the variable will be available starting from next batch
+     *                     </ul>scopes
+     * @param variableName The variable name to store results. If the variable name is my-var, the following
+     *                     variables will be created:
+     *                     <ul>
+     *                     <li>${my-var.payload} - the response from the function, or an error object</li>
+     *                     <li>${my-var.status-code} - the HTTP status code is in the 200 range for a successful
+     *                     request</li>
+     *                     <li>${my-var.log-result} - the last 4 KB of the execution log</li>
+     *                     <li>${my-var.executed-version} - the version of the function that executed, when it’s invoked
+     *                     a function with an
+     *                     alias, this indicates which version the alias resolved to</li>
+     *                     <li>${my-var.function-error} - if present, indicates that an error occurred during function
+     *                     execution, details about the error are included in the response payload</li>
+     *                     </ul>
+     */
+    @When("I invoke Lambda function `$functionName` with payload `$payload` and save result to $scopes variable "
+            + "`$variableName`")
+    public void invokeLambda(String functionName, String payload, Set<VariableScope> scopes, String variableName)
+    {
+        InvokeRequest invokeRequest = new InvokeRequest().withFunctionName(functionName).withPayload(payload);
+        InvokeResult invokeResult = awsLambdaClient.invoke(invokeRequest);
+
+        Map<String, String> result = new HashMap<>();
+        result.put("payload", new String(invokeResult.getPayload().array(), StandardCharsets.UTF_8));
+        result.put("status-code", invokeResult.getStatusCode().toString());
+        result.put("log-result",
+                new String(Base64.getDecoder().decode(invokeResult.getLogResult()), StandardCharsets.UTF_8));
+        result.put("executed-version", invokeResult.getExecutedVersion());
+        String functionError = invokeResult.getFunctionError();
+        if (functionError != null)
```

Comment:
considering the case when the user will expect an empty function error, wouldn't it be easier to verify an empty string instead of a missing variable?

Reply:
Analog of the SSH steps' empty error output stream

Reply:
- I've tested this functionality, `functionError` is some error type (https://docs.aws.amazon.com/lambda/latest/dg/invocation-retries.html), I have not found any enumeration of possible values here, I always see 'Unhandled' (in my case and in the examples), so I would prefer to rely on current default mechanism: missing value - missing variable (since there is no statement saying it couldn't be empty)

 - As for error check the users should rely on `statusCode`